packages/core/src/browser/diff-uris.ts

```diff
@@ -23,28 +23,31 @@ export namespace DiffUris {
     export const DIFF_SCHEME = 'diff';
 
     export function encode(left: URI, right: URI, label?: string): URI {
-        const diffUris = [
+        const diffUris = JSON.stringify([
             left.toString(),
             right.toString()
-        ];
+        ]);
+        const diffUriContents = JSON.stringify([
+            left.contents,
+            right.contents
+        ]);
 
-        const diffUriStr = JSON.stringify(diffUris);
-
-        return new URI().withScheme(DIFF_SCHEME).withPath(label || '').withQuery(diffUriStr);
+        return new URI().withScheme(DIFF_SCHEME).withPath(label || '').withQuery(diffUris).withContents(diffUriContents);
```

Comment:
get rid of `withContents` it is not a part of URI spec, but something which this PR invented to fix a particular issue

Reply:
Please see https://github.com/eclipse-theia/theia/pull/8589#discussion_r524977010

Reply:
@akosyakov @vince-fugnitto 
cc: @amiramw 
Reverted this changes - removed "contents" from uri.